spring-integration-ip/src/main/java/org/springframework/integration/ip/dsl/UdpOutboundChannelAdapterSpec.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.ip.dsl;
+
+import java.util.function.Function;
+
+import org.springframework.integration.dsl.MessageHandlerSpec;
+import org.springframework.integration.expression.FunctionExpression;
+import org.springframework.integration.ip.udp.UnicastSendingMessageHandler;
+import org.springframework.messaging.Message;
+
+/**
+ * A {@link MessageHandlerSpec} for {@link UnicastSendingMessageHandler}s.
+ *
+ * @author Gary Russell
+ * @since 5.0
+ *
+ */
+public class UdpOutboundChannelAdapterSpec
+		extends MessageHandlerSpec<UdpOutboundChannelAdapterSpec, UnicastSendingMessageHandler> {
+
+	protected UdpOutboundChannelAdapterSpec() {
+		super();
+	}
+
+	UdpOutboundChannelAdapterSpec(String destinationExpression) {
```

Comment:
Since there is `UnicastSendingMessageHandler(Expression destinationExpression)`, I would like to see `UdpOutboundChannelAdapterSpec(Function<Message<?>, Object> destinationFunction)`